### ISSUE TITLE
QnAMaker middleware lacks the Middleware suffix (vs LuisRecognizerMiddleware)

### ISSUE THREAD

**Issue by brandonh-msft** [Microsoft/botbuilder-dotnet#120]
Users are looking for consistency in naming to help develop and help guide development of their own middleware.

**Comment by cleemullins** [comment]
+1

Do you have opinions how how we should names these things? It seems cumbersome to continually add "Middleware" to everything, I think the price is worth the cost, as having "Middleware" makes it very obvious what's going on and (at least in .Net) the additional typing is worth is. 

On the other had, we could be **more** specific: 

1. ReceiveMiddleware
2. SendMiddleware
3. [...]

My inclination is just to use "Middlware" as the suffix. 👍

**Comment by RyanDawkins** [comment]
I agree @cleemullins. I think adding the suffix Middleware is better than without. 

This also seems to fall in line with Microsoft's other documentation on implementing common types. https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces
